### PR TITLE
Migrate the `check-protected-branch` hook to invoke

### PR DESCRIPTION
**Pull request by FlorentClarret**
### What does this PR do?

Migrate the `check-protected-branch` hook to invoke

### Motivation

- Centralize the code with `invoke`
- Test it
- Make the hooks compatible with `deva`
- People that did not load their venv and not using `deva` get the `ModuleNotFoundError` on Invoke, which is sad because it currently does not need it but it's in the init

### Additional Notes

https://datadoghq.atlassian.net/browse/ADXT-309

### CODE REVIEW

**Review by KevinFairise2 — APPROVED** [review]

File: tasks/unit-tests/git_tests.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    @unittest.mock.patch('tasks.git.get_current_branch')
+    def test_check_protected_branch_error(self, get_current_branch_mock):
+        protected_branched = (
+            'main',
+            '7.54.x',
+            '6.54.x',
+        )
+
+        for branch_name in protected_branched:
```
Comment (amenasria) [inline tasks/unit-tests/git_tests.py:27]:
```suggestion
        protected_branches = (
            'main',
            '7.54.x',
            '6.54.x',
        )

        for branch_name in protected_branches:
```

**🥜 nitpick:** ‏Not important

File: tasks/unit-tests/libs/common/git_tests.py [hunk 2]
```diff
@@ -15,3 +15,10 @@ def test_get_staged_files(self):
 
         self.assertEqual(files, ["file1", "file2", "file3"])
         self.ctx_mock.run.assert_called_once_with("git diff --name-only --staged HEAD", hide=True)
+
+    def test_get_current_branch(self):
+        self.ctx_mock.run.return_value.stdout = "  main  \n"
```
Comment (amenasria) [inline tasks/unit-tests/libs/common/git_tests.py:20]: **👏 praise:** ‏Smart

**Review by amenasria — APPROVED** [review]
LGTM !

**Pull request merged**